pkg/sources/http/http.go

```diff
@@ -159,11 +159,18 @@ func New(
 			}
 			eventTime = time.UnixMilli(i)
 		}
+
+		headers := make(map[string]string, len(r.Header))
+		for k, v := range r.Header {
+			headers[k] = v[0]
```

Comment:
Exclude headers like `Authorization`?

Reply:
Also leave a comment telling we only take the first value of the array.

Reply:
Should we pass in Auth headers, too? For example, what if some downstream vertex wants to access an external entity using the same auth headers, e.g., if the auth is BrowserAuth+ and passes the user's token for accessing the store?

Reply:
This is going to be a security concern.

The `Authorization` header is supposed to be only accessed by the client who initializes the request, and the server who receives the request, and it's encrypted during transmission. If we keep it and store it in ISB, that means the credential is leaked to and stored in a 3rd party.

Reply:
Yes. We will have to enable encryption at rest and in transit.

Reply:
opened #1583